ledger/participant-integration-api/src/main/scala/platform/index/InMemoryStateUpdater.scala

```diff
@@ -100,14 +93,43 @@ private[platform] object InMemoryStateUpdater {
     updateCachesExecutionContext = ExecutionContext.fromExecutorService(updateCachesExecutor),
     metrics = metrics,
   )(
-    updateCaches = updateCaches(inMemoryState),
-    convertTransactionAccepted = convertTransactionAccepted,
-    convertTransactionRejected = convertTransactionRejected,
-    updateLedgerEnd = updateLedgerEnd(inMemoryState),
+    prepare = prepare,
+    update = update(inMemoryState, loggingContext),
   )
 
-  private def updateCaches(inMemoryState: InMemoryState)(
-      updates: Vector[TransactionLogUpdate]
+  def extractMetadataFromUploadedPackage(batch: Vector[(Offset, Update)]): PackageMetadata =
+    batch.foldLeft(PackageMetadata()) {
+      case (metadata, (_, uploadEvent: Update.PublicPackageUpload)) =>
+        uploadEvent.archives.foldLeft(metadata) { case (acc, archive) =>
+          acc.append(PackageMetadata.from(archive))
+        }
+      case (metadata, _) => metadata
+    }
```

Comment:
```suggestion
  def extractMetadataFromUploadedPackage(batch: Vector[(Offset, Update)]): PackageMetadata =
    batch.view
      .collect { case (_, packageUpload: Update.PublicPackageUpload) => packageUpload }
      .flatMap(_.archives.view)
      .map(PackageMetadata.from)
      .foldLeft(PackageMetadata())(_ append _)
```

Reply:
this is way much nicer than my peace, thanks!